samples/cpp/benchmark_app/main.cpp

```diff
@@ -516,13 +604,43 @@ int main(int argc, char* argv[]) {
             app_inputs_info = getInputsInfo<InputInfo::CPtr>(FLAGS_shape,
                                                              FLAGS_layout,
                                                              FLAGS_b,
+                                                             FLAGS_data_shape,
                                                              FLAGS_iscale,
                                                              FLAGS_imean,
                                                              exeNetwork.GetInputsInfo());
             if (batchSize == 0) {
                 batchSize = 1;
             }
         }
+
+        if (isDynamicNetwork && FLAGS_api == "sync") {
+            throw std::logic_error("Benchmarking of the model with dynamic shapes is available for async API only."
+                                   "Please use -api async -nstreams 1 -nireq 1 to emulate sync behavior");
+        }
+
+        // Defining of benchmark mode
+        // for static models inference only mode is used as default one
+        bool inferenceOnly = FLAGS_inference_only;
+        if (isDynamicNetwork) {
+            if (isFlagSetInCommandLine("inference_only") && inferenceOnly && app_inputs_info.size() != 1) {
```

Comment:
As far as I remember, `app_inputs_info` is not about the number of tensor shapes, it is about a number of model inputs. How it is connected to the dynamic or not dynamic scenario?

Reply:
https://github.com/openvinotoolkit/openvino/blob/8d1e9155f1e169388d83a4026d72845d97f4f9aa/samples/cpp/benchmark_app/main.cpp#L462

it was changed to a vector of `benchmark_app::InputsInfo`, which contains  inputs from one tensor shape group.

Reply:
@ivikhrev Ok, I see
@AlexeyLebedev1 It seems the python tool is still not aligned with this one change.